### PR TITLE
Fixing globo #2524

### PR DESCRIPTION
**Pull request by Bricio**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixing #2524

Added 2 calls to download pages to setup cookies. First call is needed to set GLBEXP and glb_uid cookies. Second call is needed to set locksession

Changed playback video version to 2. 
Fixed the JSON format for the new version.
Added a new test case.

### CODE REVIEW

File: yt_dlp/extractor/globo.py [hunk 1]
```diff
@@ -67,11 +67,26 @@ class GloboIE(InfoExtractor):
...
+        },
+        'params': {
+            'skip_download': True,
+        },
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
+        self._download_webpage('https://globo-ab.globo.com/v2/selected-alternatives?experiments=player-isolated-experiment-02&skipImpressions=true', video_id, 'Getting cookies')
```
Comment (pukkandan) [inline yt_dlp/extractor/globo.py:88]: try this
```suggestion
        self._request_webpage(
            HEADRequest('https://globo-ab.globo.com/v2/selected-alternatives?experiments=player-isolated-experiment-02&skipImpressions=true'),
            video_id, 'Getting cookies')
```

File: yt_dlp/extractor/globo.py [hunk 2]
```diff
@@ -82,7 +97,7 @@ def _real_extract(self, url):
 
         formats = []
         security = self._download_json(
-            'https://playback.video.globo.com/v1/video-session', video_id, 'Downloading security hash for %s' % video_id,
+            'https://playback.video.globo.com/v2/video-session', video_id, 'Downloading security hash for %s' % video_id,
```
Comment (pukkandan) [inline yt_dlp/extractor/globo.py:100]: Confirm whether this works with the existing tests

File: yt_dlp/extractor/globo.py [hunk 3]
```diff
@@ -92,7 +107,10 @@ def _real_extract(self, url):
                 "tz": "-3.0:00"
             }).encode())
 
-        security_hash = security['source']['token']
+        manifest = security['sources'][0]['url_template']
+        self._download_webpage(manifest, video_id, 'Getting locksession cookie')
```
Comment (pukkandan) [inline yt_dlp/extractor/globo.py:111]:
```suggestion
        self._request_webpage(HeadRequest(security['sources'][0]['url_template']), video_id, 'Getting locksession cookie')
```

File: yt_dlp/extractor/globo.py (hunk 1 above)
Comment (Bricio) [reply yt_dlp/extractor/globo.py:88]: with this changes flake8 is returning errors.

> D:\ytdlp\yt-dlp>flake8 yt_dlp\extractor\globo.py
yt_dlp\extractor\globo.py:89:13: F821 undefined name 'HEADRequest'
yt_dlp\extractor\globo.py:112:31: F821 undefined name 'HeadRequest'

**Comment by Bricio** [comment]
Test Result:

> D:\ytdlp\yt-dlp>python test/test_download.py TestDownload.test_Globo_all
> C:\Users\User\AppData\Local\Programs\Python\Python38\lib\subprocess.py:946: ResourceWarning: subprocess 6268 is still running
>   _warn("subprocess %s is still running" % self.pid,
> ResourceWarning: Enable tracemalloc to get the object allocation traceback
> [debug] [Globo] Extracting URL: http://g1.globo.com/carros/autoesporte/videos/t/exclusivos-do-g1/v/mercedes-benz-gla-passa-por-teste-de-colisao-na-europa/3607726/
> [Globo] 3607726: Getting cookies
> [Globo] 3607726: Downloading JSON metadata
> [Globo] 3607726: Downloading security hash for 3607726
> [Globo] 3607726: Getting locksession cookie
> [Globo] 3607726: Downloading m3u8 information
> [debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
> [info] 3607726: Downloading 1 format(s): hls-2062
> [info] Writing video metadata as JSON to: test_Globo_3607726.info.json
> C:\Users\User\AppData\Local\Programs\Python\Python38\lib\subprocess.py:946: ResourceWarning: subprocess 6896 is still running
>   _warn("subprocess %s is still running" % self.pid,
> ResourceWarning: Enable tracemalloc to get the object allocation traceback
> [debug] [Globo] Extracting URL: http://globoplay.globo.com/v/4581987/
> [Globo] 4581987: Getting cookies
> [Globo] 4581987: Downloading JSON metadata
> [Globo] 4581987: Downloading security hash for 4581987
> [Globo] 4581987: Getting locksession cookie
> [Globo] 4581987: Downloading m3u8 information
> [debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
> [info] 4581987: Downloading 1 format(s): hls-2226
> [info] Writing video metadata as JSON to: test_Globo_1_4581987.info.json
> C:\Users\User\AppData\Local\Programs\Python\Python38\lib\subprocess.py:946: ResourceWarning: subprocess 2896 is still running
>   _warn("subprocess %s is still running" % self.pid,
> ResourceWarning: Enable tracemalloc to get the object allocation traceback
> [debug] [Globo] Extracting URL: https://globoplay.globo.com/v/10248083/
> [Globo] 10248083: Getting cookies
> [Globo] 10248083: Downloading JSON metadata
> [Globo] 10248083: Downloading security hash for 10248083
> [Globo] 10248083: Getting locksession cookie
> [Globo] 10248083: Downloading m3u8 information
> [debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
> [info] 10248083: Downloading 1 format(s): hls-2735
> [info] Writing video metadata as JSON to: test_Globo_2_10248083.info.json
> .
> 
> Ran 1 test in 1.495s
> 
> OK


Download Test.

> D:\ytdlp\yt-dlp>python -m yt_dlp -v "https://globoplay.globo.com/v/10248083/"
> [debug] Command-line config: ['-v', 'https://globoplay.globo.com/v/10248083/']
> [debug] Encodings: locale cp1252, fs utf-8, out utf-8 (No ANSI), err utf-8 (No ANSI), pref cp1252
> [debug] yt-dlp version 2022.01.21 [f20d607b0] (source)
> [debug] Lazy loading extractors is disabled
> [debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
> [debug] Git HEAD: a826a5d3a
> [debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
> [debug] exe versions: none
> [debug] Optional libraries: sqlite
> [debug] Proxy map: {}
> [debug] [Globo] Extracting URL: https://globoplay.globo.com/v/10248083/
> [Globo] 10248083: Getting cookies
> [Globo] 10248083: Downloading JSON metadata
> [Globo] 10248083: Downloading security hash for 10248083
> [Globo] 10248083: Getting locksession cookie
> [Globo] 10248083: Downloading m3u8 information
> [debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
> [debug] Default format spec: best/bestvideo+bestaudio
> [info] 10248083: Downloading 1 format(s): hls-2735
> [debug] Invoking downloader on "https://vod-as-03-30.video.globo.com/r90_720/v0/c8/82/df/10248083_df4d8fd36481c57f24735d8760755383392237ac/10248083-U369wJY-manifest.ism/10248083-U369wJY-manifest-audio_por=128004-video_por=2452000.m3u8?hls_client_manifest_version=4"
> [hlsnative] Downloading m3u8 manifest
> [hlsnative] Total fragments: 89
> [download] Destination: Melhores momentos - Equador 1 x 1 Brasil pelas Eliminatórias da Copa do Mundo 2022 [10248083].mp4
> [download] 100% of 169.49MiB in 00:14
> WARNING: 10248083: Possible MPEG-TS in MP4 container or malformed AAC timestamps. Install ffmpeg to fix this automatically

**Pull request merged**